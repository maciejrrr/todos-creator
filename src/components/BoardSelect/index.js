import React from 'react';
import { Select } from 'antd';
import styled from 'styled-components';

import { metrics } from '../../theme';
import NewItemForm from '../NewItemForm';

const Container = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
`;

const StyledSelect = styled(Select)`
  width: ${metrics.formWidth}px;
  margin-right: ${metrics.smallMargin}px;
`;

const BoardSelect = ({ boardChange, boards, addBoard, value }) => {
  const selectedOption = value || 'Select board';
  return (
    <Container>
      <StyledSelect
        dropdownStyle={{ width: `${metrics.formWidth}px` }}
        value={selectedOption}
        notFoundContent="No boards"
        onSelect={boardChange}
      >
        {boards.map(board => (
          <Select.Option key={board.id} value={board.id}>
            {board.name}
          </Select.Option>
        ))}
      </StyledSelect>
      <NewItemForm submit={addBoard} inline placeholder="Add a board" />
    </Container>
  );
};

export default BoardSelect;
