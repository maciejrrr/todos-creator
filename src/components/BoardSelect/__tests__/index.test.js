import BoardSelect from '../index';

describe('BoardSelect component', () => {
  let props;
  beforeEach(() => {
    props = {
      boardChange: jest.fn(),
      boards: [],
      addBoard: jest.fn(),
      value: null,
    };
  });

  describe('snapshots', () => {
    describe('no boards', () => {
      it('renders select with default message and form', () => {
        const tree = renderer.create(<BoardSelect {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });

    describe('selected board', () => {
      it('renders select with options and form', () => {
        const boards = [{ id: 1, name: 'first board' }, { id: 2, name: 'second board' }];
        const tree = renderer.create(<BoardSelect {...props} boards={boards} value={1} />).toJSON();
        expect(tree).toMatchSnapshot();
      });
    });
  });
});
