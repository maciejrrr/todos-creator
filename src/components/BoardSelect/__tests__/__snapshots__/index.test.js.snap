// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BoardSelect component snapshots no boards renders select with default message and form 1`] = `
<div
  className="sc-EHOje heLVgc"
>
  <div
    className="sc-bZQynM fMhQSn ant-select ant-select-enabled"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    onTouchStart={[Function]}
    style={undefined}
  >
    <div
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup="true"
      className="ant-select-selection
                ant-select-selection--single"
      onKeyDown={[Function]}
      role="combobox"
      tabIndex={0}
    >
      <div
        className="ant-select-selection__rendered"
      >
        <div
          className="ant-select-selection-selected-value"
          style={
            Object {
              "display": "block",
              "opacity": 1,
            }
          }
          title="Select board"
        >
          Select board
        </div>
      </div>
      <span
        className="ant-select-arrow"
        onClick={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="unselectable"
      >
        <b />
      </span>
    </div>
  </div>
  <form
    className="sc-bxivhb iGVlRn"
    onSubmit={[Function]}
  >
    <input
      className="sc-ifAKCX cMqBtW"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Add a board"
      value=""
    />
  </form>
</div>
`;

exports[`BoardSelect component snapshots selected board renders select with options and form 1`] = `
<div
  className="sc-EHOje heLVgc"
>
  <div
    className="sc-bZQynM fMhQSn ant-select ant-select-enabled"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseDown={[Function]}
    onMouseEnter={undefined}
    onMouseLeave={undefined}
    onTouchStart={[Function]}
    style={undefined}
  >
    <div
      aria-autocomplete="list"
      aria-expanded={false}
      aria-haspopup="true"
      className="ant-select-selection
                ant-select-selection--single"
      onKeyDown={[Function]}
      role="combobox"
      tabIndex={0}
    >
      <div
        className="ant-select-selection__rendered"
      >
        <div
          className="ant-select-selection-selected-value"
          style={
            Object {
              "display": "block",
              "opacity": 1,
            }
          }
          title="first board"
        >
          first board
        </div>
      </div>
      <span
        className="ant-select-arrow"
        onClick={[Function]}
        style={
          Object {
            "WebkitUserSelect": "none",
            "userSelect": "none",
          }
        }
        unselectable="unselectable"
      >
        <b />
      </span>
    </div>
  </div>
  <form
    className="sc-bxivhb iGVlRn"
    onSubmit={[Function]}
  >
    <input
      className="sc-ifAKCX cMqBtW"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Add a board"
      value=""
    />
  </form>
</div>
`;
